network/onchain-discovery/src/lib.rs

```diff
@@ -240,64 +226,18 @@ where
         }
     }
 
-    fn handle_network_event(&mut self, event: Result<Event<OnchainDiscoveryMsg>, NetworkError>) {
-        match event {
-            Ok(event) => match event {
-                Event::NewPeer(peer_id) => {
-                    trace!("connected to new peer: {}", peer_id.short_str());
-                    // Add peer to connected peer list.
-                    self.connected_peers.insert(peer_id);
-                }
-                Event::LostPeer(peer_id) => {
-                    trace!("disconnected from peer: {}", peer_id.short_str());
-                    // Remove peer from connected peer list.
-                    self.connected_peers.remove(&peer_id);
-                }
-                Event::Message(msg) => {
-                    warn!("unexpected direct-send message from network: {:?}", msg);
-                    debug_assert!(false);
-                }
-                Event::RpcRequest((peer_id, msg, res_tx)) => match msg.try_into() {
-                    Ok(OnchainDiscoveryMsg::QueryDiscoverySetRequest(req_msg)) => {
-                        debug!(
-                            "recevied query discovery set request: peer: {}, \
-                             version: {}, seq_num: {}",
-                            peer_id.short_str(),
-                            req_msg.client_known_version,
-                            req_msg.client_known_seq_num,
-                        );
-
-                        let res = self.inbound_rpc_executor.try_spawn(
-                            handle_query_discovery_set_request(
-                                Arc::clone(&self.storage_read_client),
-                                peer_id,
-                                req_msg,
-                                res_tx,
-                            ),
-                        );
-
-                        if res.is_err() {
-                            warn!(
-                                "discovery set query executor at capacity; dropped \
-                                 rpc request: peer: {}",
-                                peer_id.short_str()
-                            );
-                        }
-                    }
-                    Ok(msg) => {
-                        warn!("unexpected rpc from peer: {}, msg: {:?}", peer_id, msg);
-                        debug_assert!(false);
-                    }
-                    Err(err) => {
-                        warn!(
-                            "failed to deserialize rpc from peer: {}, err: {:?}",
-                            peer_id, err
-                        );
-                        debug_assert!(false);
-                    }
-                },
-            },
-            Err(err) => error!("network event error: {:?}", err),
+    fn handle_connection_notif(&mut self, notif: ConnectionStatusNotification) {
```

Comment:
it seems kind of wasteful to distribute peer management logic throughout the code base, is there some way to conveniently expose a read-only peer manager? Not in this PR, but maybe in another...

Reply:
i had the same thought while writing this. like it kinda sucks to keep track of these `connected_peers: HashSet<PeerId>` everywhere. but i'm not sure what the alternative looks like, since many clients might have custom logic when peer connects/disconnects.